### PR TITLE
Fix show call-out warning when there is no data in the vulnerabilities index

### PR DESCRIPTION
**Pull request by jbiset**
### Description
This pull request adds a vulnerability index data check and displays a warning call-out in case the index does not exist or there is no data in it. This applies to both the Dashboard tab and the Inventory tab.
This pull request also fixes the following changes:
- Since the architecture field will not be in the index, it is necessary to remove the architecture field - both tabs vuls
- Since the event field is not going to be in the index, it is necessary:
  - `@timestamp` is used instead of `event.create` in the "Accumulation of the most detected vulnerabilities" visualization
  - Delete `event.create` in inventory
- Capitalize is used on the values in the `vulnerability.severity` fields.
- In the OS filters of the vulnerabilities dashboard, use the `os.full` field
- Added WarningError to error handling to recognize vulnerability warnings and adjust the health check accordingly
 
### Issues Resolved
- #6166 

#### Health check warning message when no indices match vulnerability index pattern 

image

#### Vulnerability detection dashboard when no indices match vulnerability index pattern 

image

#### Vulnerability detection inventory when no indices match vulnerability index pattern 

image

#### Health check when the vulnerability index exists
image

#### Vulnerability detector dashboard when the vulnerability index exists, but mapping is wrong
image

#### Vulnerability detector dashboard when the vulnerability index exists and mapping is correct
image

#### Vulnerability detector inventory when the vulnerability index exists and mapping is correct
image

### Test
Before remember to create the mapping of port `9200:9200` in `osd1` in the `dev.yml`

- When raising the environment for the first time. If the vulnerability index does not exist, a Warning should appear in the Health Check. But it should allow you to continue. When going to the vulnerabilities detector (Dashboard and Inventory), Warning should appear because no indices match vulnerability index pattern.
- Create the index (wazuh-states-vulnerabilities) in the manager and insert data (using script of [comment](https://github.com/wazuh/wazuh-dashboard-plugins/issues/5763#issuecomment-1779564103)). The health-check will correctly create the index pattern and pass the health-check, although the dashboard will not be seen completely, because mapping is wrong.
- Insert the corresponding mapping and see that the healthcheck passes correctly and the dashboard is shown complete (using dev-tools of [comment](https://github.com/wazuh/wazuh-dashboard-plugins/issues/5763#issuecomment-1779564103)). (Refresh vulnerabilities index-pattern then). The Inventory tab should also appear complete.

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by Machi3mfl** [comment]
# Evidence

Screenshot 2023-11-30 at 15 19 10

Screenshot 2023-11-30 at 15 19 02

Screenshot 2023-11-30 at 15 18 54

Screenshot 2023-11-30 at 15 23 34


Screenshot 2023-11-30 at 15 24 06

Screenshot 2023-11-30 at 15 24 23

Screenshot 2023-11-30 at 15 24 36

**Comment by lucianogorza** [comment]
# Evidence
image
image
image

File: plugins/main/public/components/overview/vulnerabilities/dashboards/overview/dashboard_panels.ts [hunk 1]
```diff
@@ -431,7 +431,7 @@ const getVisStateAccumulationMostDetectedVulnerabilities = (
           enabled: true,
           type: 'date_histogram',
           params: {
-            field: 'event.created',
+            field: '@timestamp',
```
Comment (Desvelao) [inline plugins/main/public/components/overview/vulnerabilities/dashboards/overview/dashboard_panels.ts:434]: **thought**: the index pattern of alerts defines the `timestamp` as time filed. Should the vulnerability detection states use the same field to keep the consistency?

**Comment by Desvelao** [comment]
I reviewed the pull request changes and there is code that is doing other things that are not related to display the callout.  Maybe you could mention these changes into the description of the pull request to be aware of these too.

**Pull request merged**